src/CattyTests/Project/ProjectMigratorTest.swift

```diff
@@ -0,0 +1,144 @@
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+class ProjectMigratorTest: XCTestCase {
+
+    var newFolderName = String()
+    var header: Header!
+    var project: Project!
+
+    override func setUp() {
+        super.setUp()
+        newFolderName = kLocalizedScene
+
+        self.header = Header()
+        header.catrobatLanguageVersion = "0.992"
+
+        project = Project()
+        project.header = header
+    }
+
+    func testMigrateToScene() {
+        var automaticScreenShortAtPath = "\(String(describing: project.projectPath()))automatic_screenshot.png"
+        var imageDirectoryAtPath = "\(String(describing: project.projectPath()))images"
+        var soundDirectoryAtPath = "\(String(describing: project.projectPath()))sounds"
+
+        let fileManager = CBFileManagerMock(filePath: [automaticScreenShortAtPath], directoryPath: [imageDirectoryAtPath, soundDirectoryAtPath])
+
+        let migrate = ProjectMigrator(fileManager: fileManager)
+
+        do {
+            try migrate.migrateToScene(project: project)
+        } catch {
+            XCTFail("Some error occured")
+        }
+
+        automaticScreenShortAtPath = "\(String(describing: project.projectPath()) + newFolderName) 1/automatic_screenshot.png"
+        imageDirectoryAtPath = "\(String(describing: project.projectPath()) + newFolderName) 1/images"
+        soundDirectoryAtPath = "\(String(describing: project.projectPath() + newFolderName)) 1/sounds"
+
+        XCTAssertTrue(fileManager.fileExists(automaticScreenShortAtPath))
+        XCTAssertTrue(fileManager.directoryExists(imageDirectoryAtPath))
+        XCTAssertTrue(fileManager.directoryExists(soundDirectoryAtPath))
+    }
+
+    func testMigrateToSceneForInvalidCatrobatLanguageVersion() {
+        header.catrobatLanguageVersion = "0.991"
+        project.header = header
+
+        let fileManager = CBFileManagerMock(filePath: [] as! [String], directoryPath: [] as! [String])
+
+        let migrate = ProjectMigrator(fileManager: fileManager)
+
+        do {
+
+            try migrate.migrateToScene(project: project)
+
+        } catch let error as ProjectMigratorError {
+            XCTAssertEqual(error, ProjectMigratorError.invalidCatrobatLanguageVersion)
+        } catch {
+            XCTFail("some other error occured")
+        }
+    }
+
+    func testMigrateToSceneForFilePathNotFound() {
+        let imageDirectoryAtPath = "\(String(describing: project.projectPath()))images"
+        let soundDirectoryAtPath = "\(String(describing: project.projectPath()))sounds"
+
+        let fileManager = CBFileManagerMock(filePath: [] as! [String], directoryPath: [imageDirectoryAtPath, soundDirectoryAtPath])
+
+        let migrate = ProjectMigrator(fileManager: fileManager)
+
+        do {
+
+            try migrate.migrateToScene(project: project)
```

Comment:
Please use the Nimble framework to do that assertion (https://github.com/Quick/Nimble#swift-error-handling). This framework is already included and used in our project.